### ISSUE TITLE
hc::printf is ignoring format specifier when linking with -lmcwamp_hsa

### ISSUE THREAD

**Issue by misos1** [RadeonOpenCompute/hcc#1020]
main.cpp:

``` c++
#include <hc.hpp>

#define HCC_ENABLE_ACCELERATOR_PRINTF
#include <hc_printf.hpp>

int main()
{
	parallel_for_each(hc::extent<1>(1), [=]() [[hc]]
	{
		hc::printf("%i\n", 10);
	});
	return 0;
}
```

With "-lmcwamp_hsa":

```
$ hcc -hc main.cpp -lmcwamp_hsa
$ HCC_ENABLE_PRINTF=1 ./a.out
%i
```

Without "-lmcwamp_hsa":

```
$ hcc -hc main.cpp
$ HCC_ENABLE_PRINTF=1 ./a.out
10
```

After is added line with "processPrintfBuffer":

``` c++
#include <hc.hpp>

#define HCC_ENABLE_ACCELERATOR_PRINTF
#include <hc_printf.hpp>

int main()
{
	parallel_for_each(hc::extent<1>(1), [=]() [[hc]]
	{
		hc::printf("%i\n", 10);
	});
	hc::processPrintfBuffer(hc::printf_buffer);
	return 0;
}
```

Output:

```
$ hcc -hc main.cpp -lmcwamp_hsa
$ HCC_ENABLE_PRINTF=1 ./a.out
10
```

**Comment by ssahasra** [comment]
Reassigning to Siu Chi. I don't really know how the current printf works. On the look of it, it's rather strange that the program even compiles without "-lmcwamp_hsa", because the printf header says that the printf_buffer is defined in mcwamp_hsa.cpp.

**Comment by david-salinas** [comment]
@misos1 with latest HCC does this still occur for you?  I can't reproduce it.  Also, libmcwamp_hsa gets loaded at runtime by libmcwamp, so there really isn't any need to link it in when you compile (call hcc).

**Comment by misos1** [comment]
This is already fixed. I noticed some changes in rocm 2.3 in hc_printf.hpp where hc::processPrintfBuffer was removed. It was actually quite interesting feature to call processPrintfBuffer and flush printf buffer in middle of kernel execution, especially if it takes really long time, to print some simple progress or for debugging to see what is happening and when.

Is it not possible anymore to flush printf buffer or to call processPrintfBuffer?
